File: app/src/main/java/autogravity/com/mvvmdemo/domain/schedulers/BaseSchedulerProvider.java
package autogravity.com.mvvmdemo.domain.schedulers;

import android.support.annotation.NonNull;
import io.reactivex.Scheduler;

/**
 * Created by LeviHsiao on 4/10/18.
 */

public interface BaseSchedulerProvider {

    @NonNull Scheduler computation();

    @NonNull Scheduler io();

    @NonNull Scheduler mainThread();

    @NonNull Scheduler newThread();
}

File: app/src/main/java/autogravity/com/mvvmdemo/domain/schedulers/BaseSchedulerProvider.kt
package autogravity.com.mvvmdemo.domain.schedulers

import io.reactivex.Scheduler

/**
 * Created by LeviHsiao on 4/10/18.
 */

interface BaseSchedulerProvider {

    fun computation(): Scheduler

    fun io(): Scheduler

    fun mainThread(): Scheduler

    fun newThread(): Scheduler
}
